doc/man/pilot-install-netsync.1
.\"Generated by db2man.xsl. Don't modify this, modify the source.
.de Sh \" Subsection
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Ip \" List item
.br
.ie \\n(.$>=3 .ne \\$3
.el .ne 3
.IP "\\$1" \\$2
..
.TH "pilot-install-nets" 1 "Copyright 1996-2007 FSF" "0.12.5" "PILOT-LINK"
.SH NAME
pilot-install-netsync \- reads or sets the Network Preferences information on a Palm Device.
.SH "SECTION"

.PP
pilot\-link: Userland conduits

.SH "SYNOPSIS"

.PP
 \fIpilot\-install\-netsync\fR [\fB\-p\fR|\fB\-\-port\fR <\fBport\fR>] [\fB\-\-version\fR] [\fB\-?\fR|\fB\-\-help\fR] [\fB\-\-usage\fR] [\fB\-q\fR|\fB\-\-quiet\fR] [\fB\-e\fR|\fB\-\-enable\fR] [\fB\-d\fR|\fB\-\-disable\fR] [\fB\-n\fR|\fB\-\-name\fR  \fBname\fR] [\fB\-a\fR|\fB\-\-address\fR  \fBaddress\fR] [\fB\-m\fR|\fB\-\-mask\fR  \fBnetmask\fR]

.SH "DESCRIPTION"

.PP
 \fIpilot\-install\-netsync\fR allows the user to read or change the Network Preferences stored on the Palm\&.

.SH "OPTIONS"

.SS "pilot-install-netsync options"

                        \fB\-e\fR, \fB\-\-enable\fR
                    
.PP
Enable the LanSync operation (Used for performing a "Network" HotSync (synchronize across the Internet for example) instead of "Local" HotSync)\&.

                        \fB\-d\fR, \fB\-\-disable\fR
                    
.PP
Disable the LANSync setting on the Palm

                        \fB\-a\fR,
                        \fB\-\-address\fR=\fBaddress\fR
                    
.PP
Set the IP address of the remote machine you wish to connect to via "Network" HotSync\&. This is \fInot\fR the IP address of the Palm device itself\&. In most cases the Palm will be assigned an IP address automatically by your desktop or server\-side configuration\&.

                        \fB\-n\fR,
                        \fB\-\-name\fR \fBhostname\fR
                    
.PP
Sets the hostname of the remote machine you wish to connect to via "Network" HotSync\&. This is \fInot\fR the hostname of the Palm device itself\&.

                        \fB\-i\fR,
                        \fB\-\-ip\fR \fBip\fR
                    
.PP
This option is no longer supported in 0\&.12 \- see \-a, \-\-address\&.

                        \fB\-m\fR,
                        \fB\-\-mask\fR \fBnetmask\fR
                    
.PP
Set the netmask of the remote machine you wish to connect to via "Network" HotSync\&. This is \fInot\fR the netmask address of the Palm device itself, it should be assigned a netmask value at connection time, based on your desktop or server\-side configuration\&.

.SS "Conduit Options"

                        \fB\-p\fR, \fB\-\-port\fR lt;\fBport\fR>
                    
.PP
Use device file <\fBport\fR> to communicate with the Palm handheld\&. If this is not specified, \fIpilot\-install\-netsync\fR will look for the \fI $PILOTPORT \fR environment variable\&. If neither are found or supplied, \fIpilot\-install\-netsync\fR will print the usage information\&.

                        \fB\-q\fR, 
                        \fB\-\-quiet\fR
                    
.PP
Suppress 'Hit HotSync button' message

                        \fB\-v\fR, \fB\-\-version\fR
                    
.PP
Display version of \fIpilot\-install\-netsync\fR and exit without connecting\&.

.SS "Help Options"

                        \fB\-h\fR, \fB\-\-help\fR
                    
.PP
Display the help synopsis for \fIpilot\-install\-netsync\fR and exit without connecting\&.

                        \fB\-\-usage\fR 
                    
.PP
Display a brief usage message and exit without connecting\&.

.SH "USAGE"

.PP
The program will connect to a target Palm handheld, and read and display or set the Network Preferences information specified or display the Network Preferences stored on the Palm device\&.

.SH "EXAMPLE"

.PP
 \fB\-p\fR  \fIusb:\fR  \fB\-H\fR  \fB"NEPTUNE"\fR  \fB\-a\fR  \fB10\&.0\&.1\&.10\fR  \fB\-n\fR  \fB255\&.255\&.255\&.0\fR 

.SH "AUTHOR"

.PP
 \fIpilot\-install\-netsync\fR was created from the original \fIpilot\-install\-user\fR application originally written by D\&. Jeff Dionne <jeff@RyeHam\&.ee\&.ryerson\&.ca>, and was split into a separate conduit by JP Rosevear < jpr@ximian\&.com > \&.

.PP
This manual page was converted to XML by Neil Williams <linux@codehelp\&.co\&.uk> 

.SH "REPORTING BUGS"

.PP
We have an online bug tracker\&. Using this is the only way to ensure that your bugs are recorded and that we can track them until they are resolved or closed\&. Reporting bugs via email, while easy, is not very useful in terms of accountability\&. Please point your browser to http://bugs\&.pilot\-link\&.org: \fIhttp://bugs.pilot-link.org\fR and report your bugs and issues there\&.

.SH "COPYRIGHT"

.PP
This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 2 of the License, or (at your option) any later version\&.

.PP
This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE\&. See the GNU General Public License for more details\&.

.PP
You should have received a copy of the GNU General Public License along with this program; if not, write to the Free Software Foundation, Inc\&., 51 Franklin St, Fifth Floor, Boston, MA 02110\-1301, USA\&.

.SH "SEE ALSO"

.PP
 \fIpilot\-install\-user\fR(1), \fIpilot\-link\fR(7)\&.



src/pilot-install-netsync.c
/*
 * $Id: pilot-install-netsync.c,v 1.49 2007/06/01 01:13:55 desrod Exp $ 
 *
 * pilot-install-netsync.c:  Palm Network Information Installer
 *
 * This program is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by the
 * Free Software Foundation; either version 2 of the License, or (at your
 * option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
 * Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 *
 */

#include <stdio.h>
#include <sys/types.h>
#include <sys/socket.h>
#include <arpa/inet.h>
#include <netinet/in.h>

#include "pi-dlp.h"
#include "pi-header.h"
#include "pi-userland.h"

poptContext po;

int main(int argc, char *argv[])
{
	int 	enable	= -1,
		sd 	= -1,
		po_err	= -1;

	const char
		*hostname 	= NULL,
		*address 	= NULL,
		*netmask 	= NULL;

	const struct poptOption options[] = {
		USERLAND_RESERVED_OPTIONS
		{ "enable",  'e', POPT_ARG_VAL,    &enable, 1, "Enables LANSync on the Palm"},
		{ "disable", 'd', POPT_ARG_VAL,    &enable, 0, "Disable the LANSync setting on the Palm"},
		{ "name",    'n', POPT_ARG_STRING, &hostname, 0, "The hostname of the remote machine you sync with", "<name>"},
		{ "address", 'a', POPT_ARG_STRING, &address, 0, "IP address of the remote machine you connect to", "<address>"},
		{ "mask",    'm', POPT_ARG_STRING, &netmask, 0, "Subnet mask of the network your Palm is on", "<netmask>"},
		POPT_TABLEEND
	};

	struct 	NetSyncInfo 	Net;
	struct in_addr addr;

	po = poptGetContext("pilot-install-netsync", argc, (const char **) argv, options, 0);
	poptSetOtherOptionHelp(po," [-p port] <netsync options>\n\n"
		"   Assigns your Palm device NetSync information\n\n"
		"   Example:\n"
		"      -p usb: -e -n \"NEPTUNE\" -a 10.0.1.10 -m 255.255.255.0\n\n");

	if (argc < 2) {
		poptPrintUsage(po,stderr,0);
		return 1;
	}

	while ((po_err = poptGetNextOpt(po)) >= 0) {
	}
	if (po_err < -1) plu_badoption(po,po_err);

	/* FIXME: Take the user-supplied IP or hostname and reverse it to
	   get the other component, which reduces the complexity of this by
	   one argument passed in. getnameinfo() will help here. */
	if (address && !inet_pton(AF_INET, address, &addr)) {
		fprintf(stderr,"   ERROR: The address you supplied, '%s' is in invalid.\n"
			"      Please supply a dotted quad, such as 1.2.3.4\n\n", address);
		return 1;
	}

	if (netmask && !inet_pton(AF_INET, netmask, &addr)) {
		fprintf(stderr,"   ERROR: The netmask you supplied, '%s' is in invalid.\n"
			"      Please supply a dotted quad, such as 255.255.255.0\n\n", netmask);
		return 1;
	}

	sd = plu_connect();
	if (sd < 0)
		goto error;

	if (dlp_OpenConduit(sd) < 0)
		goto error_close;

	/* Read and write the LANSync data to the Palm device */
	if (dlp_ReadNetSyncInfo(sd, &Net) < 0)
		goto error_close;

	if (enable != -1)
		Net.lanSync = enable;

	if (!plu_quiet) {
		printf("   LANSync....: %sabled\n", (Net.lanSync == 1 ? "En" : "Dis"));
		fflush(stdout);
	}

	if (hostname)
		strncpy(Net.hostName, hostname, sizeof(Net.hostName));

	if (address)
		strncpy(Net.hostAddress, address, sizeof(Net.hostAddress));

	if (netmask)
		strncpy(Net.hostSubnetMask, netmask,
			sizeof(Net.hostSubnetMask));

	if (!plu_quiet) {
		printf("   Hostname...: %s\n", Net.hostName);
		printf("   IP Address.: %s\n", Net.hostAddress);
		printf("   Netmask....: %s\n\n", Net.hostSubnetMask);
	}

	if (dlp_WriteNetSyncInfo(sd, &Net) < 0)
		goto error_close;

	if (dlp_AddSyncLogEntry(sd, "pilot-install-netsync, exited normally.\n"
				"Thank you for using pilot-link.\n") < 0)
		goto error_close;

	if (dlp_EndOfSync(sd, 0) < 0)
		goto error_close;

	if (pi_close(sd) < 0)
		goto error;

	return 0;

error_close:
	pi_close(sd);

error:
	return -1;
}

/* vi: set ts=8 sw=4 sts=4 noexpandtab: cin */
/* ex: set tabstop=4 expandtab: */
/* Local Variables: */
/* indent-tabs-mode: t */
/* c-basic-offset: 8 */
/* End: */
